packages/legend-studio/src/components/editor/edit-panel/uml-editor/AssociationEditor.tsx

```diff
@@ -80,9 +84,25 @@ const AssociationPropertyBasicEditor = observer(
   }) => {
     const { association, property, selectProperty, isReadOnly } = props;
     const editorStore = useEditorStore();
+    const checkDuplicateValue = (val: string, id: string): void => {
+      if (
+        association.properties[0].name === val ||
+        association.properties[1].name === val
+      ) {
+        guaranteeNonNullable(
+          document.getElementById(id),
+        ).style.backgroundColor = 'red';
+      } else {
+        guaranteeNonNullable(
+          document.getElementById(id),
+        ).style.backgroundColor = 'var(--color-dark-grey-300)';
+      }
+    };
     // Name
-    const changeValue: React.ChangeEventHandler<HTMLInputElement> = (event) =>
+    const changeValue: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+      checkDuplicateValue(event.target.value, event.target.id);
```

Comment:
this is not ideal, we want this validation to be automatically done, not reactive. If the user starts with a class that already has duplicated properties, this won't work right?